include/envoy/http/filter.h

```diff
@@ -299,6 +299,11 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    * @return the buffer limit the filter should apply.
    */
   virtual uint32_t decoderBufferLimit() PURE;
+
+  /**
+   * This routine disarms the request timer owned by the connection manager, if it exists.
+   */
+  virtual void endRequestTimeout() PURE;
```

Comment:
Should this have a more generic name? For example, upstreamRequestComplete. It seems to me that the action taken by the HCM doesn't need to leak out into this interface.